controller/identity/domain.go

```diff
@@ -0,0 +1,41 @@
+package identity
+
+import (
+	"errors"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/util/validation"
+)
+
+// TrustDomain is a namespace for identities.
+type TrustDomain struct {
+	controlNS, domain string
+}
+
+// NewTrustDomain creates a new identity namespace.
+func NewTrustDomain(controlNS, domain string) (*TrustDomain, error) {
+	if errs := validation.IsDNS1123Label(controlNS); len(errs) > 0 {
+		for _, e := range errs {
+			return nil, fmt.Errorf("Invalid label '%s': %s", controlNS, e)
```

Comment:
I believe the convention in go is to have error strings not be capitalized, as they're usually embedded into other strings.